server/src/main/java/org/apache/druid/curator/CuratorModule.java

```diff
@@ -79,10 +80,33 @@ public CuratorFramework makeCurator(CuratorConfig config, EnsembleProvider ensem
           StringUtils.format("%s:%s", config.getZkUser(), config.getZkPwd()).getBytes(StandardCharsets.UTF_8)
       );
     }
+
+    RetryPolicy retryPolicy;
+    if (config.getTerminateDruidProcessOnConnectFail()) {
+      final Runnable exitRunner = () -> {
+        try {
+          log.error("Zookeeper can't be reached, forcefully stopping lifecycle...");
+          lifecycle.stop();
+          log.error("Zookeeper can't be reached, forcefully stopping virtual machine...");
```

Comment:
This should probably be `System.err.println` since `log` is probably dead after lifecycle stop. It won't explode, but it probably won't go anywhere either.

Reply:
Also done.